web/src/overlays.ts

```diff
@@ -204,3 +204,92 @@ export function initialize(): void {
         e.stopPropagation();
     });
 }
+
+export function trap_focus_for_settings_overlay(): void {
+    $("#settings_overlay_container").on("keydown", (e) => {
+        if (e.key !== "Tab") {
+            return;
+        }
+
+        const two_column_mode = $("#settings_content").css("--single-column") === undefined;
+        let visible_focusable_elements;
+        if (two_column_mode) {
+            visible_focusable_elements =
+                overlay_util.get_visible_focusable_elements_in_overlay_container(
```

Comment:
I'm a little concerned about calling this function on every `keydown` event on the overlay container when the "Tab" key has been pressed. Maybe that concern is overblown, but I could imagine a performance hit when the overlay has a lot of focusable elements, especially when a user is very quickly tabbing backwards and forwards--causing the array of focusable elements to be rebuilt multiple times in short order.

Is there a way (and material benefit) to figuring out that list when an overlay opens, and updating it when the overlay's content changes? Are there conditions under which the list of focusable elements would change from keystroke to keystroke?

Reply:
Apart from content changing, the focusable elements also change when resizing the window.

The main problem is when changing the content of overlay because that happens in multiple ways, like for tables a row can be added or removed, or a complete table re-render can take place, so there might be cases where while making some changes to the code we might miss calling the function to recompute first and last elements.

Reply:
Got it. That makes sense. Still, it would be excellent to engineer this away as part of the overlay redesign project.